warehouse/packaging/models.py

```diff
@@ -371,6 +383,7 @@ def __table_args__(cls):  # noqa
     created = Column(
         DateTime(timezone=False), nullable=False, server_default=sql.func.now()
     )
+    published = Column(DateTime(timezone=False), nullable=True)
```

Comment:
1) Well, I was following the pattern of having `created` and `updated` timestamps in models, in this case for when it was published. Then it occurred to me that instead of setting a boolean value to know if a release was published or not, we could just look at the `published` column and check for null. So we get 2 use cases for the price of 1. Otherwise, we'd need another column (or table!) to keep track of what releases are drafts...

2) This was another option that was considered, having the `draft_hash` be an attribute itself, or even a whole other model. However I think managing an attribute or another model unnecesary adds complexity to the architecture, and since I don't see most projects using this feature, I think using a `column_property` is better than having an almost-empty column. An important detail is that the idea is to only compute this value when we know that the release is a draft (by looking at the `published` column). However, I'm open to hear more in favor of having of this argument :)

3)  Yes, it's supposed to be part of the migration process, and it's automatically populated for all releases that are not explicitly a draft. So no worries there when deploying the new code.

4) You're correct. However, from what I gathered from the documentation you can't use information from other tables when using `column_property`. If you know of a way to do this, I'd be happy to avoid the denormalization :)